src/view/items/element/attribute/getUpdateDelegate.js

```diff
@@ -189,8 +189,14 @@ function updateStringValue(reset) {
 
       this.node._ractive.value = value;
 
-      this.node.value = safeToStringValue(value);
-      this.node.setAttribute('value', safeToStringValue(value));
+      const safeValue = safeToStringValue(value);
+
+      // fixes #3281 – Safari moves caret position when setting an input value to the same value
+      if (this.node.value !== safeValue) {
+        this.node.value = safeValue;
+      }
+
+      this.node.setAttribute('value', safeValue);
```

Comment:
Any reason not to include this in the conditional? Just wondering. 😁

Reply:
My thinking is that something else might have changed the attribute 'value', for example a decorator.
Like: we could have a `<input .... value="something">` that has `this.node.input === "something-else"`. In this case, we'd rather always update the "value" attribute, just to make sure.